File: Code/PS2/Matlab/qencode.m
function [out1] = qencode(ntuple, etable, multfac)
% This procedure does a quick encode of any n-tuple given in weakly
% descending order. Encoding is done using a table lookup. Each
% column of the table consists of an n-tuple; the ith column is the ith
% n-tuple to be decoded. The table is stored in the variable "etable".
out1 = etable(sum(ntuple.*multfac)+1);

end

File: Code/PS2/Python/qencode.py
def qencode(ntuple, etable, multfac):
    """
    Quickly encodes a weakly descending n-tuple using a precomputed lookup table.

    Args:
        ntuple (list): Weakly descending n-tuple.
        etable (numpy.ndarray): Encoding lookup table.
        multfac (numpy.ndarray): Multiplication factor for encoding.

    Returns:
        int: Encoded integer state code.
    """
    index = np.sum(np.array(ntuple) * np.array(multfac)) + 1  # Compute index
    return etable[index]  # Lookup encoded value
